server/tserver/src/test/java/org/apache/accumulo/tserver/util/CreateEmptyTest.java

```diff
@@ -0,0 +1,206 @@
+    replay(context);
+  }
+
+  @AfterEach
+  public void verifyMock() {
+    verify(context);
+  }
+
+  @SuppressFBWarnings(value = "PATH_TRAVERSAL_IN", justification = "path provided by test")
+  @Test
+  public void exceptionOnFileExistsTest() throws Exception {
+    CreateEmpty createEmpty = new CreateEmpty();
+
+    String wal1 = genFilename(tempDir.getAbsolutePath() + "/empty", ".wal");
+    String rf1 = genFilename(tempDir.getAbsolutePath() + "/empty", ".rf");
+
+    // create the file so it exists
+    File f = new File(wal1);
+    assertTrue(f.createNewFile());
+
+    String[] walArgs = {"--type", "WAL", wal1};
+    CreateEmpty.Opts walOpts = new CreateEmpty.Opts();
+    walOpts.parseArgs("accumulo create-empty", walArgs);
+
+    assertThrows(IllegalArgumentException.class,
+        () -> createEmpty.createEmptyWal(walOpts, context));
+
+    // create the file so it exists
+    File f2 = new File(rf1);
+    assertTrue(f2.createNewFile());
+
+    String[] rfArgs = {"--type", "RF", rf1};
+    CreateEmpty.Opts rfOpts = new CreateEmpty.Opts();
+    rfOpts.parseArgs("accumulo create-empty", rfArgs);
+    assertThrows(IllegalArgumentException.class,
+        () -> createEmpty.createEmptyRFile(walOpts, context));
+  }
+
+  @Test
+  public void createRfileTest() throws Exception {
+    CreateEmpty createEmpty = new CreateEmpty();
+
+    String file1 = genFilename(tempDir.getAbsolutePath() + "/empty", ".rf");
+    String file2 = genFilename(tempDir.getAbsolutePath() + "/empty", ".rf");
+
+    String[] args = {"--type", "RF", file1, file2};
+    CreateEmpty.Opts opts = new CreateEmpty.Opts();
+    opts.parseArgs("accumulo create-empty", args);
+
+    createEmpty.createEmptyRFile(opts, context);
+    VolumeManager vm = context.getVolumeManager();
+    assertTrue(vm.exists(new Path(file1)));
+    assertTrue(vm.exists(new Path(file2)));
+  }
+
+  /**
+   * Validate that the default type is RF (RecoveryWithEmptyRFileIT also needs this(
+   */
+  @Test
+  public void createRfileDefaultTest() throws Exception {
+    CreateEmpty createEmpty = new CreateEmpty();
+
+    String file1 = genFilename(tempDir.getAbsolutePath() + "/empty", ".rf");
+
+    String[] args = {file1};
+    CreateEmpty.Opts opts = new CreateEmpty.Opts();
+    opts.parseArgs("accumulo create-empty", args);
+
+    createEmpty.createEmptyRFile(opts, context);
+    VolumeManager vm = context.getVolumeManager();
+    assertTrue(vm.exists(new Path(file1)));
+  }
+
+  @Test
+  public void createWalTest() throws Exception {
+    CreateEmpty createEmpty = new CreateEmpty();
+
+    String file1 = genFilename(tempDir.getAbsolutePath() + "/empty", ".wal");
+    String file2 = genFilename(tempDir.getAbsolutePath() + "/empty", ".wal");
+
+    String[] args = {"--type", "WAL", file1, file2};
+    CreateEmpty.Opts opts = new CreateEmpty.Opts();
+    opts.parseArgs("accumulo create-empty", args);
+
+    createEmpty.createEmptyWal(opts, context);
+
+    checkWalContext(file1);
```

Comment:
In addition to this check could call `org.apache.accumulo.tserver.logger.LogReader` to ensure these files can be read.  LogReader is another user facing util.

However, not sure if its easily possible to do that.  If it seems like a huge amount of work, could open a follow on issue or ignore this comment.  This would be easy from the command line with accumulo installed as both create empty and log reader can be executed using the accumulo command, but may not be easy to do in a test.

Reply:
In 4d34d80fe6 I added an additional scan of the wal file (based on what was in LogReader) but not sure it adds much to what was already there.

Reply:
> In https://github.com/apache/accumulo/commit/4d34d80fe66ad9451ca6ae4a7751e0f1bbdfd473 I added an additional scan of the wal file (based on what was in LogReader) but not sure it adds much to what was already there.

I like what was added there. If uses the same code that LogSorter uses to open the wal and sort it, so that is a good test for regression purposes.